homeassistant/components/heos/config_flow.py

```diff
@@ -20,15 +20,22 @@ class HeosFlowHandler(config_entries.ConfigFlow):
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+    DISCOVERED_HOSTS = {}
```

Comment:
This dictionary will be shared across all instances of ConfigFlow, create a dict inside `__init__` instead. Example: https://github.com/home-assistant/home-assistant/blob/7a6950fd72edf6d716de5ce8b7f0c5bde573cacb/homeassistant/components/hue/config_flow.py#L50-L52

Reply:
This is an intentional class instance.  Because discovery creates a new instance of the flow handler for each request, we need to gather these together into a single list.

Reply:
We could store it in hass.data?

Reply:
We could -- but it's only used within the flow handler -- is that better than just using a class attribute?

Reply:
I would avoid putting multiple objects on the class. It's easy to be misinterpreted.

Reply:
Moved to `hass.data`